data/stylish-haskell.yaml

```diff
@@ -294,6 +294,97 @@ steps:
       # Default: false
       post_qualify: false
 
+      # Automatically group imports based on their module names, with
+      # a blank line separating each group. Groups are ordered in
+      # alphabetical order.
+      #
+      # By default, this groups by the first part of each module's
+      # name (Control.* will be grouped together, Data.*... etc), but
+      # this can be configured with the group_patterns setting.
+      #
+      # When enabled, this rewrites existing blank lines and groups.
+      #
+      # - true: Group imports by the first part of the module name.
+      #
+      #   > import Control.Applicative
+      #   > import Control.Monad
+      #   > import Control.Monad.MonadError
+      #   >
+      #   > import Data.Functor
+      #
+      # - false: Keep import groups as-is (still sorting and
+      #   formatting the imports within each group)
+      #
+      #   > import Control.Monad
+      #   > import Data.Functor
+      #   >
+      #   > import Control.Applicative
+      #   > import Control.Monad.MonadError
+      #
+      # Default: false
+      group_imports: false
+
+      # A list of rules specifying how to group modules and how to
+      # order the groups.
+      #
+      # Each rule has a match field; the rule only applies to module
+      # names matched by this pattern.
+      #
+      # Rules are processed in order, so only the *first* rule that
+      # matches a specific module will apply. Any module names that do
+      # not match a single rule will be put into a single group at the
+      # end of the import block.
+      #
+      # Example: group MyApp modules first, with everything else in
+      # one group at the end.
+      #
+      #  group_rules:
+      #    - match: "^MyApp\\>"
```

Comment:
```suggestion
      #    - match: "^MyApp\\."
```

Typo?

Reply:
Oh, `\\>` is an anchor for the end of the word. `^MyApp\\>` would match the module name `MyApp` while `^MyApp\\.` wouldn't.

I had a note about that in the documentation at one point, but looks like I lost it in one of the revisions. I'll add a note about it after the bit about ^ and $, since it's a handy feature from POSIX regexps that is different in PCRE (which uses `\b` instead) and that people might not know about.

Reply:
Awesome, yeah, I only knew about `\b`!